pkg/deployments/tasks/20230217-gnosis-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,326 @@
+    const veBALTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    veBAL = await veBALTask.instanceAt('VotingEscrow', veBALTask.output({ network: 'mainnet' }).VotingEscrow);
+
+    bal80weth20Pool = await deployedAt('v2-pool-weighted/WeightedPool', VEBAL_POOL);
+
+    const poolId = await bal80weth20Pool.getPoolId();
+
+    await vault.connect(veBALHolder).joinPool(
+      poolId,
+      veBALHolder.address,
+      veBALHolder.address,
+      {
+        assets: [BAL, ZERO_ADDRESS],
+        maxAmountsIn: [0, VAULT_BOUNTY],
+        fromInternalBalance: false,
+        userData: WeightedPoolEncoder.joinExactTokensInForBPTOut([0, VAULT_BOUNTY], 0),
+      },
+      { value: VAULT_BOUNTY }
+    );
+
+    await bal80weth20Pool.connect(veBALHolder).approve(veBAL.address, MAX_UINT256);
+    const currentTime = await currentTimestamp();
+    await veBAL
+      .connect(veBALHolder)
+      .create_lock(await bal80weth20Pool.balanceOf(veBALHolder.address), currentTime.add(MONTH * 12));
+  });
+
+  it('create gauge', async () => {
+    const tx = await factory.create(recipient.address, weightCap);
+    const event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+
+    gauge = await task.instanceAt('GnosisRootGauge', event.args.gauge);
+
+    expect(await factory.isGaugeFromFactory(gauge.address)).to.be.true;
+    expect(await gauge.getRecipient()).to.equal(recipient.address);
+  });
+
+  it('grant permissions', async () => {
+    // We need to grant permission to the admin to add the Optimism factory to the GaugeAdder, and also to then add
+    // gauges from said factory to the GaugeController.
+    const govMultisig = await impersonate(GOV_MULTISIG, fp(100));
+
+    const selectors = ['addGaugeFactory', 'addGnosisGauge'].map((method) => gaugeAdder.interface.getSighash(method));
+    await Promise.all(
+      selectors.map(
+        async (selector) =>
+          await authorizer.connect(govMultisig).grantRole(await gaugeAdder.getActionId(selector), admin.address)
+      )
+    );
+
+    // We also need to grant permissions to mint in the gauges, which is done via the Authorizer Adaptor
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(await authorizerAdaptor.getActionId(gauge.interface.getSighash('checkpoint')), admin.address);
+
+    // Currently we only have Gauge types defined up to Arbitrum. We need to add the Gnosis GaugeType to be able to add the factory
+    // First grant permission to call add_type to the admin.
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gaugeController.interface.getSighash('add_type(string,uint256)')),
+        admin.address
+      );
+
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gaugeController.interface.getSighash('add_gauge(address,int128)')),
+        admin.address
+      );
```

Comment:
There's no support for gnosis in the adder, and it currently doesn't work. We've also tested the adder throughly, so there's no point in doing that again here. Just add it manually by calling `gc.add_gauge`.